src/cdk-experimental/column-resize/resizable.ts

```diff
@@ -0,0 +1,233 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterViewInit,
+  ElementRef,
+  Inject,
+  Injector,
+  NgZone,
+  OnDestroy,
+  Type,
+  ViewContainerRef
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {ComponentPortal, PortalInjector} from '@angular/cdk/portal';
+import {Overlay, OverlayRef} from '@angular/cdk/overlay';
+import {CdkColumnDef} from '@angular/cdk/table';
+import {merge, ReplaySubject} from 'rxjs';
+import {filter, takeUntil} from 'rxjs/operators';
+
+import {closest} from '../popover-edit/polyfill';
+
+import {HEADER_ROW_SELECTOR} from './constants';
+import {ResizeOverlayHandle} from './overlay-handle';
+import {ColumnResize} from './column-resize';
+import {ColumnSizeAction, ColumnResizeNotifierSource} from './column-resize-notifier';
+import {HeaderRowEventDispatcher} from './event-dispatcher';
+import {ResizeRef} from './resize-ref';
+import {ResizeStrategy} from './resize-strategy';
+
+export abstract class Resizable<HandleComponent extends ResizeOverlayHandle>
+    implements AfterViewInit, OnDestroy {
+  protected minPxInternal: number = 0;
+  protected maxPxInternal: number = Number.MAX_SAFE_INTEGER;
+
+  protected inlineHandle?: HTMLElement;
+  protected overlayRef?: OverlayRef;
+  protected readonly destroyed = new ReplaySubject<void>();
+  protected readonly document: Document;
+
+  get minPx(): number {
```

Comment:
How impractical would it be to support more units for minx/max?

Reply:
It would ultimately require the ability to convert them into pixels for use while resizing is in progress. I'm not going to do it, but someone probably could.